SSA/Projects/InstCombine/AliveAutoGenerated.lean

```diff
@@ -720,37 +720,35 @@ theorem alive_AndOrXor_144  (w : Nat)   : alive_AndOrXor_144_src w  ⊑ alive_An
   %r = icmp eq %a3, 0
 
 -/
--- MANUAL FIX: the translation script inferred some of the resulting types to be `i1`
--- where they should be the meta-variable `_` (https://github.com/opencompl/ssa/issues/169)
-def alive_AndOrXor_698_src  (w : Nat)   :=
-[alive_icom ( w )| {
+def alive_AndOrXor_698_src   :=
+[alive_icom ()| {
 ^bb0(%a : _, %b : _, %d : _):
-  %v1 = llvm.and %a, %b
-  %v2 = llvm.and %a, %d
-  %v3 = llvm.mlir.constant 0
-  %v4 = llvm.icmp.eq %v1, %v3
-  %v5 = llvm.mlir.constant 0
-  %v6 = llvm.icmp.eq %v2, %v5
+  %v1 = llvm.and %a, %b : i1
+  %v2 = llvm.and %a, %d : i1
+  %v3 = llvm.mlir.constant 0 : _
+  %v4 = llvm.icmp.eq %v1, %v3 : i1
+  %v5 = llvm.mlir.constant 0 : _
+  %v6 = llvm.icmp.eq %v2, %v5 : i1
```

Comment:
This is the first one that breaks. @goens, any idea why the `(w : Nat)` is being removed?

This code is working:

```
def alive_AndOrXor_698_src (w : Nat)  :=
[alive_icom (w)| {
^bb0(%a : _, %b : _, %d : _):
  %v1 = llvm.and %a, %b : _
  %v2 = llvm.and %a, %d : _
  %v3 = llvm.mlir.constant 0 : _
  %v4 = llvm.icmp.eq %v1, %v3 : _
  %v5 = llvm.mlir.constant 0 : _
  %v6 = llvm.icmp.eq %v2, %v5 : _
  %v7 = llvm.and %v4, %v6 : i1
  llvm.return %v7 : i1
}]

def alive_AndOrXor_698_tgt (w : Nat) :=
[alive_icom (w)| {
^bb0(%a : _, %b : _, %d : _):
  %v1 = llvm.or %b, %d : _
  %v2 = llvm.and %a, %v1 : _
  %v3 = llvm.and %a, %b : _
  %v4 = llvm.and %a, %d : _
  %v5 = llvm.mlir.constant 0 : _
  %v6 = llvm.icmp.eq %v3, %v5 : _
  %v7 = llvm.mlir.constant 0 : _
  %v8 = llvm.icmp.eq %v4, %v7 : _
  %v9 = llvm.mlir.constant 0 : _
  %v10 = llvm.icmp.eq %v2, %v9 : _
  llvm.return %v10 : i1
}]
```

Reply:
is this being generated with the python script? In this case, the heuristic we use to find out wether to add the w broke with the "pretty" syntax and it seems to still not work on all cases. I think this might relate to the unification bug.

Reply:
Tobias Grosser: Right, this is generated with the python script. I tried to make the PR as simple as possible. There is also a choice for us to drop the ": _" in case no type is given, to further reduce the diff (or me adding it back on main).

Tobias Grosser: For some reason, alive marks the and over a, b and d to have bitwidth 1, yet the block arguments still have type _.

Tobias Grosser: Maybe it droppes the w because the uses in the and are marked as i1?